### PR TITLE
Shadow Rifts

### PR DESCRIPTION
**Pull request by gausie**
Adds:

* Shadow Rift location
* New shadow monsters
* New items and effects
* Tracking for the point of ingress to the rift

Ingress point is also reflected in Location Details and `appearance_rates`

### CODE REVIEW

File: src/data/monsters.txt [hunk 1]
```diff
@@ -583,6 +583,18 @@ serialbus	982	serialbus.gif	Atk: 60 Def: 54 HP: 60 Init: 50 Meat: 30 P: demon E:
 Servant of Grodstank	799	grodseal.gif	NOCOPY FREE Atk: 175 Def: 157 HP: 200 Init: -10000 P: horror EA: stench Article: a	fustulent seal grulch (35)	infernal toilet brush (5)
 sewer snake with a sewer snake in it	1752	sewersnake.gif	Atk: 2 Def: 1 HP: 2 Init: -10000 P: beast SNAKE E: stench Article: a	carton of snake milk (100)	carton of snake milk (30)	sewer snake (30)
 shadow of Black Bubbles	798	shadowseal.gif	NOCOPY FREE Atk: 175 Def: 157 HP: 200 Init: -10000 P: horror EA: spooky Article: a	scrap of shadow (35)	shadowy seal eye (5)
+shadow bat	2271	shadowbat.gif	Atk: 100 Def: 100 HP: 100 Init: 300 P: horror Article: a	shadow sausage (5)
```
Comment (midgleyc) [inline src/data/monsters.txt:586]: all these guys are `EA: cold EA: spooky`. prism is also `EA: hot`.

Is the drop rate being 5% for every monster a dev leak? If not good job getting all 12 within the day!

Comment (gausie) [reply src/data/monsters.txt:586]: It's not a dev leak, it was just the current theory. Figured it was better than nothing - would you rather a (0)? What normally means unspaded here?

Comment (midgleyc) [reply src/data/monsters.txt]: yeah (0) means unspaded.

It probably is better than nothing but I'm curious as to what it's based on. Did somebody PFC one of the drops on a copied monster? Turns divided by items? Ran autumn-aton on the rift with one acuity, no drops, two acuity, some drops? Just eyeballing, eh, seems reasonable?

File: src/net/sourceforge/kolmafia/request/FightRequest.java [hunk 2]
```diff
@@ -10468,6 +10468,18 @@ else if (responseText.contains("Rock Promoters are long gone")) {
...
           break;
+        case ItemPool.SHADOW_BRICK:
+          if (responseText.contains("They collide, and are annihilated")) {
+            itemSuccess = true;
+          }
+      }
+    }
+
+    if (itemSuccess || itemRunawaySuccess) {
+      var limit = DailyLimitType.USE.getDailyLimit(itemId);
```
Comment (midgleyc) [inline src/net/sourceforge/kolmafia/request/FightRequest.java:10479]: the only thing using this currently is the shadow brick, is that right?

File: src/net/sourceforge/kolmafia/request/PlaceRequest.java [hunk 3]
```diff
@@ -138,262 +128,211 @@ public static void parseResponse(final String urlString, final String responseTe
...
-        if (responseText.contains("You show Jeremy the big snail shell you found")) {
-          Preferences.setBoolean("drippyShieldUnlocked", true);
-          ResultProcessor.removeItem(ItemPool.DRIPPY_SNAIL_SHELL);
-        }
-        // Oooh, that big fingernail, that's interesting.
-        // I'll send it up to the Armory right away.
-        // In the meantime, our scouts have cleared the way to another section of the Drip.
-        // There's some kind of big building there. Go check it out, would you?
-        if (responseText.contains("Oooh, that big fingernail, that's interesting")) {
+    switch (place) {
```
Comment (midgleyc) [inline src/net/sourceforge/kolmafia/request/PlaceRequest.java:131]: next time you do a large-scale refactoring like this, can you do it in a separate commit (or even PR)? It's hard to see the relevant changes.

Also can this be a switch expression? I don't see any fall-through.

File: src/net/sourceforge/kolmafia/request/FightRequest.java (hunk 2 above)
Comment (gausie) [reply src/net/sourceforge/kolmafia/request/FightRequest.java:10479]: correct!

File: src/net/sourceforge/kolmafia/request/PlaceRequest.java (hunk 3 above)
Comment (gausie) [reply src/net/sourceforge/kolmafia/request/PlaceRequest.java:131]: Yeah I wanted to do that and forgot that I had it sitting there when I came back to work more, so it was sort of locked in without me repeating work.

Comment (gausie) [reply src/net/sourceforge/kolmafia/request/PlaceRequest.java:131]: What do you mean by your second question?

File: src/net/sourceforge/kolmafia/request/PlaceRequest.java [hunk 4]
```diff
@@ -474,6 +417,12 @@ public static boolean registerRequest(final String urlString) {
               default -> null;
             };
       }
+      case "beanstalk" -> {
+        if (action.equals("stalk-rift")) {
+          message = "Entering the Shadow Rift via the The Misspelled Cemetary";
```
Comment (midgleyc) [inline src/net/sourceforge/kolmafia/request/PlaceRequest.java:422]: beanstalk, not cemetary

File: src/net/sourceforge/kolmafia/request/PlaceRequest.java [hunk 5]
```diff
@@ -489,9 +438,15 @@ public static boolean registerRequest(final String urlString) {
...
-        if (action.equals("cem_advtomb")) {
-          message = "The Unknown Tomb";
-          turns = true;
+        switch (action) {
+          case "cem_advtomb" -> {
+            message = "The Unknown Tomb";
+            turns = true;
+          }
+          case "cem_shadowrift" -> {
+            message = "Entering the Shadow Rift via the The Misspelled Cemetary";
```
Comment (midgleyc) [inline src/net/sourceforge/kolmafia/request/PlaceRequest.java:447]: the The

File: src/net/sourceforge/kolmafia/request/PlaceRequest.java (hunk 3 above)
Comment (midgleyc) [reply src/net/sourceforge/kolmafia/request/PlaceRequest.java:131]: it's currently `switch` with `case:`. It could be `switch` with `case ->`, and that would save me having to track every case to ensure there's a `break`.

File: src/net/sourceforge/kolmafia/request/PlaceRequest.java (hunk 5 above)
Comment (gausie) [reply src/net/sourceforge/kolmafia/request/PlaceRequest.java:447]: a thematically appropriate zone to typo

File: src/net/sourceforge/kolmafia/request/PlaceRequest.java (hunk 4 above)
Comment (gausie) [reply src/net/sourceforge/kolmafia/request/PlaceRequest.java:422]: thanks

File: src/net/sourceforge/kolmafia/request/PlaceRequest.java (hunk 3 above)
Comment (gausie) [reply src/net/sourceforge/kolmafia/request/PlaceRequest.java:131]: Done! Only didn't do it that way because IntelliJ didn't offer it to me automatically haha

**Review by midgleyc — APPROVED** [review]
